### PR TITLE
feat(swarm): add `ConnectionHandler::connection_keep_alive` default implementation

### PR DESCRIPTION
**Pull request by leonzchang**
## Description

Add `ConnectionHandler::connection_keep_alive` default implementation.

Related: #4306.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by thomaseizinger** [comment]
Great! This definitely needs a changelog entry and we need to adjust the docs as well to explain that this is optional.

File: swarm/src/handler.rs [hunk 1]
```diff
@@ -125,25 +124,23 @@ pub trait ConnectionHandler: Send + 'static {
 
     /// Returns until when the connection should be kept alive.
```
Comment (thomaseizinger) [inline swarm/src/handler.rs:125]: I think we need to completely overhaul this together with https://github.com/libp2p/rust-libp2p/pull/4595. What do you think of merging this PR into https://github.com/libp2p/rust-libp2p/pull/4595? It is so closely related, I think it makes sense to just merge it as a single-patch.

Comment (leonzchang) [reply swarm/src/handler.rs:125]: Sure, what do your think we also change `KeepAlive` to boolean in #4595

Comment (thomaseizinger) [reply swarm/src/handler.rs:125]: > Sure, what do your think we also change `KeepAlive` to boolean in #4595

I think it makes sense. With only two variants, we don't really gain anything from having another type and I cannot see how somebody could misinterpret what `true` and `false` mean in this case.

**Pull request closed without merging**